burn-core/src/nn/loss/ctc.rs

```diff
@@ -7,6 +7,8 @@ use burn_tensor::{backend::Backend, Element, ElementConversion, Int, Numeric, Te
 use super::Reduction;
 
 const NEG_INF: f32 = -1e5;
+// a small value used to prevent the occurrence of log(0)
+const DELTA: f32 = -1e-5;
```

Comment:
Did you mean to have this number as negative? The literal number you has is positive. In an unlikely event,  `(l1 - m.clone()).exp() + (l2 - m.clone()).exp()` expression could be equal to `abs(DELTA)` which would still lead to log(0) situation. 

Additionally, I suggest we use [https://doc.rust-lang.org/std/primitive.f32.html#associatedconstant.EPSILON](f32's EPSILON) or [f16's EPSILON]https://docs.rs/tract-core/latest/tract_core/prelude/struct.f16.html#associatedconstant.EPSILON constants depending what on Backend's precision settings. @nathanielsimard  or @louisfd can suggest on how we can extract this.   `-1e-5` seems a rather big number for f16 or f32. (probably it may not work for f16 because its epsilon is `4.88e-04`. we need to double check it)

Reply:
I'm sorry, it's a typo. `DELTA` should be positive.

`1e-5` can ensure that the results of the loss are accurate to three decimal places, but `4.88e-4` is a bit large. Perhaps CTC Loss is indeed not suitable for the use of half-precision training.